server/api/Products.js

```diff
@@ -1,46 +1,89 @@
 const {
-  models: { Product },
+  models: { Product, Room },
 } = require('../db/index');
 const router = require('express').Router();
 
 // Select all products
 
-router.get('/', async (req, res, next) => {
+router.get('/all', async (req, res, next) => {
   try {
     allProducts = await Product.findAll();
-
     res.status(200).send(allProducts);
+  } catch (err) {
+    next(err);
+  }
+});
+
+// get with queries to paginate and filter
+
+router.get('/', async (req, res, next) => {
+  const { page, size, type, style, room, sort } = req.query;
+  const options = {
+    attributes: {
+      include: [
+        'id',
+        'name',
+        'material',
+        'color',
+        'imageUrl',
+        'price',
+        'type',
+        'style',
+        'availability',
+      ],
+    },
+    order: ['name'],
+    where: {},
+  };
+  if (page && size) {
+    options.offset = (page - 1) * size;
+    options.limit = size;
+  }
```

Comment:
If the query specifies a `page` and `size`, we set the `offset` and `limit` properties of the options object